ipam/allocator.go

```diff
@@ -51,9 +52,11 @@ type Allocator struct {
 	nicknames        map[router.PeerName]string   // so we can map nicknames for rmpeer
 	pendingAllocates []operation                  // held until we get some free space
 	pendingClaims    []operation                  // held until we know who owns the space
+	dead             map[string]time.Time         // containers we heard were dead, and when
```

Comment:
Alternatively, you could use a heap, and make the removeDeadContainers significantly cheaper (wouldn't have to iterate through all the recently died) and make reclaiming the oldest dead ip O(1).